clients/roscpp/include/ros/timer.h

```diff
@@ -68,7 +68,7 @@ class ROSCPP_DECL Timer
   /**
    * \brief Set the period of this timer
    */
-  void setPeriod(const Duration& period);
+  void setPeriod(const Duration& period, const bool reset=true);
```

Comment:
For a plain `bool` argument `const` does not make much sense. Please remove it from all signatures. Please also add documentation for the new parameter.